lib/dicom_net/association.ex

```diff
@@ -109,6 +111,60 @@ defmodule DicomNet.Association do
     {new_state, response}
   end
 
+  defp reject_association(state, association_data) do
+    new_state =
+      state
+      |> Map.put(:state, :association_release_request)
+      |> Map.put(:association, association_data)
+
+    response =
+      Pdu.new_association_release_response_pdu()
+      |> Pdu.serialize()
+
+    {new_state, response}
+  end
+
+  defp handle_pdu(
+         %Pdu{
+           type: :associate_request,
+           data: associate_request
+         },
+         %{socket: _socket, state: :waiting_for_association} = state
+       ) do
+    Logger.info("Accepting association")
```

Comment:
This is now a bit misleading. Imagine a normal user of this library is just displaying info-level logs and above. Now they would see "Accepting association", but then the association could actually be rejected. I think I would remove this one, and promote your debug logs below to info, as association establishment is quite important for a DICOM server.